aws_lambda_powertools/tracing/tracer.py

```diff
@@ -177,6 +178,31 @@ def __init__(
         if self._is_xray_provider():
             self._disable_xray_trace_batching()
 
+    def set_attribute(self, key: str, value: Union[str, numbers.Number, bool]):
+        """Set attribute on current active trace entity with a key-value pair.
+
+        Parameters
+        ----------
+        key : str
+            attribute key
+        value : Union[str, numbers.Number, bool]
+            Value for attribute
+
+        Example
+        -------
+        Set attribute for a pseudo service named payment
+
+            tracer = Tracer(service="payment")
+            tracer.set_attribute("PaymentStatus", "CONFIRMED")
+        """
+        if self.disabled:
+            logger.debug("Tracing has been disabled, aborting set_attribute")
+            return
+
+        logger.debug(f"setting attribute on key '{key}' with '{value}'")
+
+        self.provider.set_attribute(key=key, value=value)
```

Comment:
I believe since no `kwargs` are being passed here, all  "attributes" will be treated as metadata, and never as annotation. Is this the correct behaviour?

```python
    def set_attribute(self, key: str, value: str | Number | bool, **kwargs) -> None:
        if kwargs.get("namespace", "") != "":
            self.put_metadata(key=key, value=value, namespace=kwargs["namespace"])
        else:
            self.put_annotation(key=key, value=value)
```

Reply:
changed the behavior of x_ray provider's `set_attribute`. Now namespace will be passed in.